Fix crash when removing scenery groups

Gymnasiast opened the pull request:
Reported by posix. I haven't yet heard if I can make the park public, so I won't.

IntelOrca commented:
Where does the crash happen? Can we not also protect it from invalid selection IDs?

Gymnasiast commented:
The crash happens in Scenery.cpp. Initially, I introduced extra null checks, but then it crashed in another place. Then I introduced them there as well, and had it crash in another place. It's why I decided to go down this route.
